pyiceberg/manifest.py

```diff
@@ -281,7 +363,6 @@ class DataFile(Record):
     split_offsets: Optional[List[int]]
     equality_ids: Optional[List[int]]
     sort_order_id: Optional[int]
-    spec_id: Optional[int]
```

Comment:
Removed this for now. Created an issue: https://github.com/apache/iceberg/issues/8712